### DISCUSSION TITLE
Equivalent of `make  -t` (GNU Make touch)

### DISCUSSION THREAD

**Discussion by grantmcdermott** [ropenscilabs/drake-manual#172] [Q&A] [answered]
Apologies if I missed something obvious, but does drake have an equivalent of GNU Make's "touch" flag (`-t`)? See description [here](https://www.gnu.org/software/make/manual/html_node/Instead-of-Execution.html#Instead-of-Execution).

My situation is that I need to make a (minor) change to an upstream function that will have no impact on my existing targets --- it's an additional function argument that defaults to NULL --- but is needed for a new target that I want to create. To avoid rebuilding everything, I'd like to introduce my minor change and then update the timestamps of my existing targets without rebuilding. Only then would I create the new target that relies on this minor change.

P.S. The "skip_targets" argument looks like it *might* be what I'm looking for, but it's not entirely clear from the description. I think that this could just be a temporary build shortcut, rather than a way of updating the target timestamps.

**Comment by wlandau** [comment, marked as answer]
Sounds like `drake` triggers might help: https://books.ropensci.org/drake/triggers.html. `make(plan, ..., trigger = trigger(depend = FALSE))` allows you to change a function without invalidating downstream targets, but this will also disable all other instances of dependency-induced invalidation. Alternatively, `make(plan..., trigger = trigger(condition = FALSE, mode = "condition"))` should not rerun any targets unless they have not been built yet.
